x-pack/legacy/plugins/index_management/public/components/mappings_editor/reducer.ts

```diff
@@ -113,7 +115,20 @@ export const reducer = (state: State, action: Action): State => {
         },
       };
     case 'documentField.changeEditor':
-      return { ...state, documentFields: { ...state.documentFields, editor: action.value } };
+      const switchingToDefault = action.value === 'default';
+      const fields = switchingToDefault ? normalize(state.fieldsJsonEditor.format()) : state.fields;
+      const nextState = {
+        ...state,
+        fields,
+        documentFields: { ...state.documentFields, editor: action.value },
+      };
+
+      if (switchingToDefault) {
+        nextState.fieldsJsonEditor.isValid = undefined;
+      } else {
+        if (nextState.fieldForm) nextState.fieldForm.isValid = undefined;
```

Comment:
I think we can set the `fieldForm` to undefined when switching? And it'd be good to reset the ```

So the nextState would be something like

```js
const nextState: State = {
  ...state,
  fields,
  fieldForm: undefined,
  documentFields: {
    ...state.documentFields,
    status: 'idle',
    fieldToAddFieldTo: undefined,
    fieldToEdit: undefined,
    editor: action.value,
  },
};
```

And just return that.